website/docs/basics/blocks-intro.md

```diff
@@ -195,4 +195,32 @@ And folder structure `src/Blocks/blocks` is kinda weird.
 
 You can, but you shouldn't. We use this key to provide the CSS selectors on all our custom blocks in the block editor. Additional styles are added to this selector for the wrapper to work on full width and remove some of the native block editor styles. 
 
-So don't remove or change this 😅 .
+So don't remove or change this 😅.
+
+### Simple vs Compound blocks
+
+In a nutshell, there is no difference between blocks. The WordPress core block is a block, but we like to make a distinction between what is simple and what is a compound block.
+
+**Simple block** is a block that is used just as is and provides a small functionality like heading, paragraph, button, etc.
+
+**Compound block** is a block that builds from multiple components and provides a bigger functionality.
+
+### Naming is hard
+
+*There are only two hard things in Computer Science: cache invalidation and naming things. - Phil Karlton*
+
+Yes, naming is hard, and no matter how long is your development experience you will always struggle with names for your components, block, files, variables, functions, etc.
+
+Here are some of our recommendations to ease your pain:
+* You should never name your component based on its usage. For example, you have a component card for custom post-type books, you should never call this component `card-book` bat rather use a more generic name like `card-product`.
+* Make your names as generic as possible for better reusability.
```

Comment:
```suggestion
* Make your names as generic as possible for better reusability, but specific enough to fully understand what the block/component is used for.
```

I would suggest to add this part. Not sure if everyone agrees but for me it seems useful.